Mohamed_Sahal/BinaryTree/Runner/BinaryTree.java

```diff
@@ -0,0 +1,114 @@
+
+import java.util.ArrayList;
+
+public class BinaryTree {
+
+    public static void main(String[] args) {
+        City seattle  =  new City("Seattle", 0);
+        City denver  =  new City("denver", 1305);
+        City miami  =  new City("miami", 2064);
+        City LosAngelos  =  new City("Los Angelos", 1016);
+        City dallas  =  new City("dallas", 1435);
+        City minneapols  =  new City("minneapolis", 990);
+        City newyork  =  new City("newyork", 2401);
+        City boston  =  new City("boston", 216);
+        //Node
+        Node root = new Node(seattle);
+
+        // Left Childen
+
+        root.left = new Node(denver);
+        root.left.left = new Node(miami);
+        root.left.right = new Node(LosAngelos);
+        root.left.right.left = new Node(dallas);
+        root.left.right.left.right = new Node(minneapols);
+
+        // Right
+
+        root.right = new Node(newyork);
+        root.right.right = new Node(boston);
+
+
+        BinaryTree.FindPath(root, "minneapolis");;
+        computeDistance();
+
+//        FindPathIterativly(root);;
+
+
+
+
+    }
+
+      /*
+        Recursive Solution
+     */
+
+    public static ArrayList<Object> RouteMiles = new ArrayList<>();
```

Comment:
Global functions are not good practice unless they are necessary. In this case they are not. Also, instead of passing `ArrayList<Object>` you could've passed `ArrayList<Integer>`. Better readability. Also, why `RouteMiles` instead of `routeMiles`? Java is camelCase.